api/v1alpha3/types.go

```diff
@@ -321,6 +323,42 @@ type UserAssignedIdentity struct {
 	ProviderID string `json:"providerID"`
 }
 
+const (
+	// AzureIdentityBindingSelector is the label used to match with the AzureIdentityBinding
+	// For the controller to match an identity binding, it needs a [label] with the key `aadpodidbinding`
+	// whose value is that of the `selector:` field in the `AzureIdentityBinding`.
+	AzureIdentityBindingSelector = "capz-controller-aadpodidentity-selector"
+)
+
+// IdentityType represents different types of identities.
+type IdentityType string
+
+const (
+	// UserAssignedMSI represents a user-assigned identity.
+	UserAssignedMSI IdentityType = "UserAssignedMSI"
+
+	// ServicePrincipal represents a service principal.
+	ServicePrincipal IdentityType = "ServicePrincipal"
+)
+
+// IdentitySpec specifies the parameters that are used to create an AzureIdentity
+type IdentitySpec struct {
+	// Name is the name of the identity
+	Name string `json:"name"`
+	// UserAssignedMSI or Service Principal
+	Type IdentityType `json:"type"`
+	// User assigned MSI resource id.
+	// +optional
+	ResourceID string `json:"resourceID,omitempty"`
+	// Both User Assigned MSI and SP can use this field.
+	ClientID string `json:"clientID"`
+	// Used for service principal
+	// +optional
+	ClientPassword corev1.SecretReference `json:"clientPassword,omitempty"`
```

Comment:
```suggestion
	ClientSecret corev1.SecretReference `json:"clientSecret,omitempty"`
```

These are usually referred to as "Client Secrets" in UI and doc https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#option-2-create-a-new-application-secret. Using password, might make it seem like a only the text secret is usable rather than certificate secret as well.